src/register_hall_of_fame.c

```diff
@@ -0,0 +1,3067 @@
+    AddTextPrinterParameterizedWithColor(&windows[2], 0, data->strbuf2, 2, 0, TEXT_SPEED_NOTRANSFER, MAKE_TEXT_COLOR(15, 2, 0), NULL);
+
+    switch (RegisterHallOfFame_GetMetLocationType(data, mon, data->args->profile)) {
+    case REGHOF_METLOC_CAUGHT:
+        ReadMsgDataIntoString(data->msgData, msg_0180_00007, data->strbuf1);
+        BufferLandmarkName(data->msgFormat, 0, hofMon->metLocation);
+        StringExpandPlaceholders(data->msgFormat, data->strbuf2, data->strbuf1);
+        break;
+    case REGHOF_METLOC_HATCHED:
+        ReadMsgDataIntoString(data->msgData, msg_0180_00008, data->strbuf1);
+        BufferLandmarkName(data->msgFormat, 0, hofMon->metLocation);
+        StringExpandPlaceholders(data->msgFormat, data->strbuf2, data->strbuf1);
+        break;
+    case REGHOF_METLOC_TRADED:
+        ReadMsgDataIntoString(data->msgData, msg_0180_00009, data->strbuf2);
+        break;
+    case REGHOF_METLOC_KANTO:
+        ReadMsgDataIntoString(data->msgData, msg_0180_00010, data->strbuf2);
+        break;
+    case REGHOF_METLOC_HOENN:
+        ReadMsgDataIntoString(data->msgData, msg_0180_00011, data->strbuf2);
+        break;
+    case REGHOF_METLOC_SINNOH:
+        ReadMsgDataIntoString(data->msgData, msg_0180_00014, data->strbuf2);
+        break;
+    case REGHOF_METLOC_ORRE:
+        ReadMsgDataIntoString(data->msgData, msg_0180_00012, data->strbuf2);
+        break;
+    case REGHOF_METLOC_FATEFUL:
+        ReadMsgDataIntoString(data->msgData, msg_0180_00013, data->strbuf2);
+        break;
+    case REGHOF_METLOC_SHUCKIE:
+    case REGHOF_METLOC_KENYA:
+        ReadMsgDataIntoString(data->msgData, msg_0180_00015, data->strbuf1);
+        BufferLandmarkName(data->msgFormat, 0, hofMon->metLocation);
+        StringExpandPlaceholders(data->msgFormat, data->strbuf2, data->strbuf1);
+        break;
+    }
+    AddTextPrinterParameterizedWithColor(&windows[2], 0, data->strbuf2, 2, 24, TEXT_SPEED_NOTRANSFER, MAKE_TEXT_COLOR(15, 2, 0), NULL);
+    CopyWindowPixelsToVram_TextMode(&windows[2]);
+}
+
+static void RegisterHallOfFame_IndivMonsScene_CreateMonSpriteAndEnableObjLayer(RegisterHallOfFameData *data, int monIdx) {
+    RegisterHallOfFame_IndivMonsScene_CreateMonSprites(data, monIdx);
+    GfGfx_EngineATogglePlanes(GX_PLANEMASK_OBJ, GF_PLANE_TOGGLE_ON);
+}
+
+static void RegisterHallOfFame_IndivMonsScene_DeleteSprites(RegisterHallOfFameData *data) {
+    for (u32 i = 0; i < 15; ++i) {
+        RegisterHallOfFame_DeletePicByIndex(data, i);
+    }
+    RegisterHallOfFame_IndivMonsScene_UnloadMonOverworldSprite(data);
+    RegisterHallOfFame_IndivMonsScene_UnloadSpriteGfx(data);
+}
+
+static void RegisterHallOfFame_IndivMonsScene_CreateMonSprites(RegisterHallOfFameData *data, int monIdx) {
+    RegisterHofMon *hofMon = &data->mons[monIdx];
+    RegisterHallOfFame_IndivMonsScene_LoadSpriteGfx(data);
+    RegisterHallOfFame_IndivMonsScene_LoadMonOverworldSprite(data, hofMon);
+    for (int i = 0; i <= 5u; ++i) {
```

Comment:
are you sure this is an int and not a u32?

Reply:
iirc i tried building this (or another function with a similar pattern to this) declaring `i` as `u32`, and it didn't match